File: python/scrabble-score/scrabble_score.py
VALUES = {
    1:  ('A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'),
    2:  ('D', 'G'),
    3:  ('B', 'C', 'M', 'P'),
    4:  ('F', 'H', 'V', 'W', 'Y'),
    5:  ('K'),
    8:  ('J', 'X'),
    10: ('Q', 'Z')
}

LETTER_VALUES = {l: v for v in VALUES.keys() for l in VALUES[v]}


def score(word):
    return sum(LETTER_VALUES[l] for l in word.upper())

File: typescript/scrabble-score/scrabble-score.ts
const letterValues = {
  a: 1,
  b: 3,
  c: 3,
  d: 2,
  e: 1,
  f: 4,
  g: 2,
  h: 4,
  i: 1,
  j: 8,
  k: 5,
  l: 1,
  m: 3,
  n: 1,
  o: 1,
  p: 3,
  q: 10,
  r: 1,
  s: 1,
  t: 1,
  u: 1,
  v: 4,
  w: 4,
  x: 8,
  y: 4,
  z: 10,
} as const;

type Letter = keyof typeof letterValues;

const hasScore = (ch: string): ch is Letter => {
  return ch in letterValues;
};

export default (word?: string): number => {
  if (word === undefined) {
    return 0;
  }

  return [...word.toLowerCase()].reduce(
    (score, ch) => score + (hasScore(ch) ? letterValues[ch] : 0),
    0,
  );
};
